demo/infra/app/app.tf

```diff
@@ -79,42 +93,91 @@ resource "kubernetes_manifest" "vault-dynamic-secret" {
       ]
     }
   }
+
+  field_manager {
+    # force field manager conflicts to be overridden
+    force_conflicts = true
+  }
 }
 
-resource "kubernetes_manifest" "vault-dynamic-secret-create" {
+resource "kubernetes_manifest" "templates" {
   manifest = {
     apiVersion = "secrets.hashicorp.com/v1beta1"
-    kind       = "VaultDynamicSecret"
+    kind       = "SecretTransformation"
     metadata = {
-      name      = "vso-db-demo-create"
+      name      = "vso-templates"
       namespace = kubernetes_namespace.dev.metadata[0].name
     }
     spec = {
-      namespace = vault_auth_backend.default.namespace
-      mount     = vault_database_secrets_mount.db.path
-      path      = local.db_creds_path
-      destination = {
-        create : true
-        name : "vso-db-demo-created"
+      templates = {
+        "app.props" = {
+          name = "app.props"
```

Comment:
How is the name of a `template` used?

Reply:
The template name is used when creating a new `template.Template` and during template execution. It acts as an accessor to some template data.

ref: https://pkg.go.dev/text/template#New and https://pkg.go.dev/text/template#New